File: leetcode/136-single-number/SingleNumber.cpp
#include "bits/stdc++.h"
using namespace std;

int singleNumber(vector<int>& nums) {
    int ans = 0;
    for (int x : nums) {
        ans ^= x;
    }
    return ans;
}

int main(int argc, char const *argv[]) {
    {
        cout << "=====Example 1=====\n";
        vector<int> nums {2, 2, 1};
        int result = singleNumber(nums);
        int correct_result = 1;
        cout << "Expected:\n\t" << correct_result << "\n";
        cout << "Output:\n\t" << result << "\n";
    }
    {
        cout << "=====Example 2=====\n";
        vector<int> nums {4, 1, 2, 1, 2};
        int result = singleNumber(nums);
        int correct_result = 4;
        cout << "Expected:\n\t" << correct_result << "\n";
        cout << "Output:\n\t" << result << "\n";
    }
    return 0;
}

File: leetcode/260-single-number-3/single_number.py
import collections
from typing import List


def singleNumber(nums: List[int]) -> List[int]:
    counter = collections.Counter(nums)
    return [num for num, cnt in counter.items() if cnt == 1]


if __name__ == "__main__":
    nums = [1, 2, 1, 3, 2, 5]
    result = singleNumber(nums)
    correct_result = [3, 5]
    print("Expected:")
    print(correct_result)
    print("Output:")
    print(result)
